File: packages/dotnet/Ygo74.AgentRuntime/Domains/Streaming/StreamTermination.cs
using Ygo74.AgentRuntime.Domains.Contracts;

namespace Ygo74.AgentRuntime.Domains.Streaming;

public static class StreamTermination
{
    public static StreamingEvent Completion(string requestId, int sequence, object finalOutput)
        => new(requestId, sequence, "completion", null, finalOutput);

    public static StreamingEvent Error(string requestId, int sequence, ErrorEnvelope error)
        => new(requestId, sequence, "error", null, null, error);
}

File: packages/java/ygo74-agent-runtime/src/main/java/com/ygo74/agentruntime/domains/streaming/StreamTermination.java
package com.ygo74.agentruntime.domains.streaming;

import com.ygo74.agentruntime.domains.contracts.ErrorEnvelope;
import com.ygo74.agentruntime.domains.contracts.StreamingEvent;

public final class StreamTermination {
    private StreamTermination() {}

    public static StreamingEvent completion(String requestId, int sequence, Object finalOutput) {
        return new StreamingEvent(requestId, sequence, "completion", null, finalOutput, null);
    }

    public static StreamingEvent error(String requestId, int sequence, ErrorEnvelope error) {
        return new StreamingEvent(requestId, sequence, "error", null, null, error);
    }
}
